MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/ProductServiceDefault.java

```diff
@@ -229,6 +229,27 @@ public CancellableTask getProductTags(int page, Callback<List<String>> callback)
     }
 
     @Override
+    public CancellableTask getProducts(final int page, final Set<String> tags, final Callback<List<Product>> callback) {
+        return new CancellableTaskSubscriptionWrapper(getProducts(page, tags).subscribe(new InternalCallbackSubscriber<>(callback)));
+    }
+
+    @Override
+    public Observable<List<Product>> getProducts(final int page, final Set<String> tags) {
+        if (page < 1) {
+            throw new IllegalArgumentException("page is a 1-based index, value cannot be less than 1");
+        }
+
+        final String tagsQueryStr = tags != null && !tags.isEmpty() ? TextUtils.join(",", tags.toArray()) : null;
```

Comment:
Sorry confused: "As well, then you could avoid the need to create the empty set to pass in for the other getProducts(page) call." where do you see the call with empty set?